### PR TITLE
Add execution role parameter to AddStepsOperator

### PR DESCRIPTION
**Pull request by syedahsn**
EMR launched a new [feature](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-steps-runtime-roles.html) which allows configuring runtime roles for EMR steps. This PR allows users to use this feature with the `EmrAddStepsOperator`. System test has been updated to use this feature.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/emr.py [hunk 1]
```diff
@@ -113,8 +122,14 @@ def execute(self, context: Context) -> list[str]:
         steps = self.steps
         if isinstance(steps, str):
             steps = ast.literal_eval(steps)
-
-        return emr_hook.add_job_flow_steps(job_flow_id=job_flow_id, steps=steps, wait_for_completion=True)
+        # from pdb import set_trace
+        # set_trace()
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/operators/emr.py]: Debugging code you forgot to delete?

(Also fyi you can use `breakpoint()` for the same effect without importing pdb)

Comment (syedahsn) [reply airflow/providers/amazon/aws/operators/emr.py]: Oops, that's embarrassing 😅. But I didn't know about the `breakpoint()` method, so it looks like I learned something 😃

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by uranusjr** [comment]
Needs to resolve conflicts.

**Comment by syedahsn** [comment]
Made a minor change with how the new parameter gets used in the hook. Because we are using `@mock_emr` decorator in some of the tests, it applies boto3 validation rules on the parameters passed, which include no `None` values, as well as a minimum length (otherwise I could have done a simpler `ExecutionRoleArn=execution_role_arn or ""`).

**Comment by Taragolis** [comment]
>Made a minor change with how the new parameter gets used in the hook. Because we are using @mock_emr decorator in some of the tests, it applies boto3 validation rules on the parameters passed, which include no None values, as well as a minimum length (otherwise I could have done a simpler ExecutionRoleArn=execution_role_arn or "").

I think this originally comes from `botocore` service definition, example for input of [AddJobFlowStepsInput](https://github.com/boto/botocore/blob/e5a2878a90d28e77d09126579e9f131a4611fb3d/botocore/data/emr/2009-03-31/service-2.json#L775-L794).

I've also found that if set some of optional parameters set to `None` or empty string then `botocore` do not propagate it to API call but sometimes it raise an error `¯\_(ツ)_/¯`

File: tests/providers/amazon/aws/operators/test_emr_add_steps.py [hunk 2]
```diff
@@ -224,4 +224,5 @@ def test_wait_for_completion(self, mock_add_job_flow_steps):
             job_flow_id=job_flow_id,
             steps=[],
             wait_for_completion=False,
+            execution_role_arn=None,
         )
```
Comment (Taragolis) [inline tests/providers/amazon/aws/operators/test_emr_add_steps.py:228]: Could we also have a test scenario in Hook?
1. Provide value
2. Provide empty value: `None`, `""`

**Review by Taragolis — APPROVED** [review]

**Pull request merged**